lib/style-transform.js

```diff
@@ -1,407 +1,45 @@
-                                        } else if (chr === ':') {
-                                          // pseudo-class or preudo-element
-                                          _line += piece + suffix
-                                          piece = chr
-                                          inCombinatorOrPseudo = true
-                                        } else if (chr === '+' || chr === '~' || chr === '>') {
-                                          // combinators
-                                          piece += chr
-                                          inCombinatorOrPseudo = true
-                                        } else {
-                                          piece += chr
-                                        }
-                                      }
-                                    }
-                                  }
-                                }
-
-                                // flush remainder
-                                if (piece.length) {
-                                  _line += piece
-                                  if (!inGlobal && !inCombinatorOrPseudo) {
-                                    _line += suffix
-                                  }
-                                }
-
-                                if (isLast) {
-                                  _line += '{'
-                                } else {
-                                  _line += ','
-                                }
-                            }
-
-                            line = _line;
-                        }
-                    }
-                }
-
-                generator && generator.addMapping({
-                    generated: {
-                      line: 1,
-                      column: output.length
-                    },
-                    source: fileName,
-                    original: {
-                        line: lineNumber,
-                        column: columnNumber
-                    }
-                })
-
-                output += line;
-                line = '';
-            }
-            // not `\t`, `\r`, `\n` characters
-            else if (code !== 9 && code !== 13 && code !== 10) {
-                line += styles[i];
-            }
-
-            // next character
-            i++;
-            columnNumber++;
+const stylis = require('stylis')
+
+function middleware(ctx, str, line, col, prefix) {
+if (ctx === 1.5) {
+  // stylis does not prefix `:global()`
+  if (str.indexOf(prefix) === 0 && str.indexOf(' ') !== -1) {
+    // remove prefix
+    str = str.substring(prefix.length).trim();
+
+    var indexOf;
+
+    // single selector
+    if (str.indexOf(' ') !== -1) {
+      return str.replace(/^[^ \t]*?\[.*?\]|[^ \n\r\t\+\~\>]+|.+\]/g, function (match) {
+        // :hover etc...
+        if ((indexOf = match.indexOf(':')) !== -1) {
+          // :global()
+          if (match.charCodeAt(indexOf+1) === 103) {
```

Comment:
i tend to lean towards`charCodeAt` since it's faster to compare numbers, i'll add the necessary comments so it's clear what character it's looking for.